tests/kubernetes/minikube.py

```diff
@@ -0,0 +1,204 @@
+from contextlib import contextmanager
+from functools import partial as p
+from kubernetes import config as kube_config
+from tests.helpers.util import *
+from tests.kubernetes.agent import Agent
+from tests.kubernetes.utils import *
+import docker
+import os
+import time
+import yaml
+
+MINIKUBE_VERSION = os.environ.get("MINIKUBE_VERSION", "v0.26.1")
+K8S_SERVICES_DIR = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'services')
+
+class Minikube:
+    def __init__(self):
+        self.container = None
+        self.client = None
+        self.version = None
+        self.name = None
+        self.host_client = docker.from_env(version="auto")
+        self.yamls = []
+        self.agent = Agent()
+        self.cluster_name = "minikube"
+        self.kubeconfig = None
+        self.namespace = "default"
+        self.ip = None
+
+    def get_client(self):
+        if self.container:
+            self.client = docker.DockerClient(base_url="tcp://%s:2375" % self.container.attrs["NetworkSettings"]["IPAddress"], version='auto')
+            return self.client
+        else:
+            return None
+
+    def get_ip(self):
+        if self.container:
+            self.ip = self.container.attrs["NetworkSettings"]["IPAddress"]
+            return self.ip
+        else:
+            return None
+
+    def load_kubeconfig(self, kubeconfig_path="/kubeconfig", timeout=300):
+        assert wait_for(p(container_cmd_exit_0, self.container, "test -f %s" % kubeconfig_path), timeout_seconds=timeout), "timed out waiting for the minikube cluster to be ready!\n\n%s\n\n" % self.container.logs().decode('utf-8').strip()
+        self.kubeconfig = "/tmp/scratch/kubeconfig-%s" % self.container.id[:12]
+        time.sleep(2)
+        rc, output = self.container.exec_run("cp -f %s %s" % (kubeconfig_path, self.kubeconfig))
+        assert rc == 0, "failed to get %s from minikube!\n%s" % (kubeconfig_path, output.decode('utf-8'))
+        time.sleep(2)
+        kube_config.load_kube_config(config_file=self.kubeconfig)
+
+    def connect(self, name, timeout, version=None):
```

Comment:
What exactly is this connecting to?

Reply:
See https://github.com/signalfx/signalfx-agent/pull/295#discussion_r189635230.